proxy/src/main/java/com/velocitypowered/proxy/connection/util/ServerListPingHandler.java

```diff
@@ -44,11 +47,29 @@ public ServerListPingHandler(VelocityServer server) {
 
   private ServerPing constructLocalPing(ProtocolVersion version) {
     VelocityConfiguration configuration = server.getConfiguration();
+    List<ServerPing.SamplePlayer> samplePlayers;
+    if (configuration.getSamplePlayersInPing()) {
+      List<ServerPing.SamplePlayer> unshuffledPlayers = server.getAllPlayers().stream()
+              .map(p -> {
+                if (p.getPlayerSettings().isClientListingAllowed()) {
+                  return new ServerPing.SamplePlayer(p.getUsername(), p.getUniqueId());
+                } else {
+                  return new ServerPing.SamplePlayer(
+                          "Anonymous Player",
+                          new UUID(0L, 0L)
+                  );
+                }
+              })
+              .collect(Collectors.toList());
+      Collections.shuffle(unshuffledPlayers);
+      samplePlayers = unshuffledPlayers.stream().limit(12).collect(Collectors.toList());
```

Comment:
There's no need for streams here, see [`List#subList`](https://docs.oracle.com/javase/8/docs/api/java/util/List.html#subList-int-int-). Ideally we would use Fisher-Yates in place (i.e. Durnstenfield) to avoid shuffling the entire list, but I couldn't find any implementation on the JDK (the `Collections.shuffle` javadoc hints at using this method, but doesn't allow stopping at a given index) nor Guava. We can evaluate the performance hit on larger servers and see if it's worth the extra method.